pretix_servicefees/signals.py

```diff
@@ -141,14 +141,19 @@ def get_fees(
         if split_taxes:
             # split taxes based on products ordered
             d = defaultdict(lambda: Decimal("0.00"))
+            trs = {}
             for p in positions:
                 if isinstance(p, CartPosition):
                     tr = p.item.tax_rule
                 else:
                     tr = p.tax_rule
-                d[tr] += p.price - p.tax_value
+                if not tr:
+                    tr = tax_rule_zero
+                key = (tr.rate, tr.code)
+                d[key] += p.price - p.tax_value
+                trs[key] = tr
```

Comment:
Is this necessary to make it work for products with no tax rule? I'd think the `if not tr:` block would be enough?

This part seems to change the behavior slightly, so that tax rules are grouped by (rate, code). If that's a change that is intended, we should note it in the commit message.

Reply:
You are right, this was actually intended as usually there is only one tax_rule with the same rate and code. But the more I think about it, we probably should keep it the way it was. The problem was `tax_rule_zero` not being hashable due to being a „Model instances without primary key“. But for the has we could just use the `tr.pk` instead. That would allow us to use tax_rule_zero for products without any tax-rule and still keep the original behaviour to explicitly distinguish between tax-rules with the same rate and code.

TLDR; I will change it to tr.pk.